frontend/src/App.tsx

```diff
@@ -2,25 +2,34 @@ import React, { FunctionComponent, ReactElement } from "react";
 import { Route, Switch } from "react-router-dom";
 import { Redirect } from "react-router";
 
+import { useRecoilValue } from "recoil";
 import { UploadPage } from "./pages/upload/UploadPage";
 import { LibraryPage } from "./pages/library/LibraryPage";
 import { NotFound } from "./components/NotFound/NotFound";
+import { modalDialogAtom } from "./store/atoms";
+import { CreateMethodologyDialog } from "./components/dialogs/CreateMethodologyDialog";
+import { CreateProjectDialog } from "./components/dialogs/CreateProjectDialog";
 
 export const App: FunctionComponent = (): ReactElement => {
+  const modalDialog = useRecoilValue(modalDialogAtom);
   return (
-    <Switch>
-      <Route path="/upload">
-        <UploadPage />
-      </Route>
-      <Route path="/library">
-        <LibraryPage />
-      </Route>
+    <>
+      {modalDialog === "create-methodology" && <CreateMethodologyDialog />}
+      {modalDialog === "create-project" && <CreateProjectDialog />}
```

Comment:
Why they are here?

Reply:
well... there are several ways to handle modals, and this one seems to be one of them. 
the idea that each of these dialogs can be showed in different places, so they are kind of global. 
so they are here to avoid code duplication in different components, but have single point of handling

Reply:
you can google and find several ways:
1. dispatching JSX (suppressing warning about not serialisable object in redux)
2. using react portals (afaiu  with code duplication)
3. this one (I've seen the similar way in one library, it has `ModalRoot` component which contains several dialogs as children, and a modal is shown depending on which id is dispatched)